pulsar-broker-common/src/main/java/org/apache/pulsar/broker/web/AuthenticationFilter.java

```diff
@@ -76,8 +77,15 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
                 // not sasl type, return role directly.
                 String role = authenticationService.authenticateHttpRequest((HttpServletRequest) request);
                 request.setAttribute(AuthenticatedRoleAttributeName, role);
-                request.setAttribute(AuthenticatedDataAttributeName,
-                    new AuthenticationDataHttps((HttpServletRequest) request));
+                String authMethodName = httpRequest.getHeader("X-Pulsar-Auth-Method-Name");
+                if (authMethodName != null && authenticationService.getAuthenticationProvider(authMethodName) != null) {
+                    AuthenticationState authenticationState = authenticationService
+                            .getAuthenticationProvider(authMethodName).newHttpAuthState(httpRequest);
+                    request.setAttribute(AuthenticatedDataAttributeName, authenticationState.getAuthDataSource());
```

Comment:
This change will break the `FunctionApiResource#clientAuthData` method. We'll need to update that method's casting and return signature from `AuthenticationDataHttps` to `AuthenticationDataSource`.

https://github.com/apache/pulsar/blob/8cb4c10fb705859efc68b4f556061abf14172d2c/pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/FunctionApiResource.java#L57-L59

This is also true in the `PulsarWebResource` class.

Reply:
Fixed

Reply:
@gaoran10 I think it's because here `request.setAttribute` accepts an `AuthenticationDataSource` rather than a `AuthenticationDataHttps`.

Reply:
Yes, but it seems that the implementation of `AuthenticationDataSource` here will always be `AuthenticationDataHttps`.

Reply:
Got it, thanks.